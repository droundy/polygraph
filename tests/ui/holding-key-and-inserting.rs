polygraph::schema!{
    type Database;
    pub struct Int(pub usize);
}

fn main() {
    let mut db = Database::<bool>::new().unwrap();
    let fortytwo = db.insert_int(Int(42));
    assert_eq!(fortytwo.0, 42);
    let fifty = db.insert_int(Int(50));
    assert_eq!(fortytwo.0, 42);
    assert_eq!(fifty.0, 50);
}
