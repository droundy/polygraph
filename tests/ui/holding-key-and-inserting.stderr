error[E0499]: cannot borrow `db` as mutable more than once at a time
  --> $DIR/holding-key-and-inserting.rs:10:17
   |
8  |     let fortytwo = db.insert_int(Int(42));
   |                    -- first mutable borrow occurs here
9  |     assert_eq!(fortytwo.0, 42);
10 |     let fifty = db.insert_int(Int(50));
   |                 ^^ second mutable borrow occurs here
11 |     assert_eq!(fortytwo.0, 42);
   |     --------------------------- first borrow later used here
   |
   = note: this error originates in a macro outside of the current crate (in Nightly builds, run with -Z external-macro-backtrace for more info)
